sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/TypeUtilsSuite.scala

```diff
@@ -43,4 +43,28 @@ class TypeUtilsSuite extends SparkFunSuite {
     typeCheckPass(ArrayType(StringType, containsNull = true) ::
       ArrayType(StringType, containsNull = false) :: Nil)
   }
+
+  test("compareBinary") {
+    def oldVersion(x: Array[Byte], y: Array[Byte]): Int = {
```

Comment:
This test case looks not useful to me. It's also confusing. We don't know what old version means without the context of this PR.

Since this is a refactoring work, if we have enough tests for `compareBinary`, passing these tests sounds good enough. If we don't have, it is good time to add some.

Reply:
fixed